### PR TITLE
Fixes the way images URLs are computed

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

This fixes a few issues in the image URLs computing:

* The registry had to be copied from the global flags to the `ImageFlags`. This is quite error prone and already some images didn't have it.
* Don't use a different way to compute images in the proxy and the server tools.

This PR changes the `ComputeImage()` function to pass the registry and flags as parameters to avoid mistakes and move the tags computation there.

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added

- [ ] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/24849

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: shared/utils/utils.go [hunk 1]
```diff
@@ -166,16 +166,24 @@ func YesNo(question string) (bool, error) {
...
 	name := imageFlags.Name
-	if !strings.Contains(imageFlags.Name, imageFlags.Registry) {
-		name = path.Join(imageFlags.Registry, path.Base(imageFlags.Name))
-		log.Info().Msgf(L("The image name provided is %[1]s and does not contains the registry %[2]s. The image name used will be %[3]s. You can set the flag --registry to change this behaviour."), imageFlags.Name, imageFlags.Registry, name)
+	if !strings.Contains(imageFlags.Name, registry) {
+		name = path.Join(registry, path.Base(imageFlags.Name))
+		log.Info().Msgf(
+			L(`The image name provided is %[1]s and does not contains the registry %[2]s.
+			The image name used will be %[3]s, set the flag --registry to change this behaviour.`),
+			imageFlags.Name, registry, name)
```
Comment (aaannz) [inline shared/utils/utils.go:179]: This is `Info` level message so displayed by default. I think that is quite chatty for that. Can't we just say `Using registry %s` without copying image path and help part?

File: shared/utils/utils.go [hunk 2]
```diff
@@ -166,16 +166,24 @@ func YesNo(question string) (bool, error) {
 }
 
 // ComputeImage assembles the container image from its name and tag.
-func ComputeImage(imageFlags types.ImageFlags, appendToName ...string) (string, error) {
-	if !strings.Contains(DefaultNamespace, imageFlags.Registry) {
-		log.Info().Msgf(L("Registry %[1]s would be used instead of namespace %[2]s"), imageFlags.Registry, DefaultNamespace)
+func ComputeImage(registry string, globalTag string, imageFlags types.ImageFlags, appendToName ...string) (string, error) {
```
Comment (aaannz) [inline shared/utils/utils.go:169]: It's not clear to me why having extra `registry` string when registry is already part of `imageFlags`?

Comment (aaannz) [reply shared/utils/utils.go:169]: Or rather, why removing registry from imageFlags?

File: shared/utils/utils.go (hunk 1 above)
Comment (cbosdo) [reply shared/utils/utils.go:180]: I'm not convinced about the usefulness of this log either... I just reformatted it.

**Comment by rjmateus** [comment]
We need to check how this PR will interfere with the logic merged on this PR: https://github.com/uyuni-project/uyuni-tools/pull/381

We should keep the logic that was implemented in that one, and defined in: https://github.com/uyuni-project/uyuni-tools/issues/378

**Comment by cbosdo** [comment]
> We need to check how this PR will interfere with the logic merged on this PR: #381
> 
> We should keep the logic that was implemented in that one, and defined in: #378

I completely dropped the commit splitting the `--registry` parameter in two: I'm washing my hands off that decision...

**Review by aaannz — APPROVED** [review]
LGTM.

For clarification, this PR also contains this one, right? https://github.com/uyuni-project/uyuni-tools/pull/392/

**Comment by cbosdo** [comment]
> LGTM.
> 
> For clarification, this PR also contains this one, right? #392

Yes it does. I can try to remove that commit though

**Pull request closed without merging**